src/igmrf.jl

```diff
@@ -171,7 +177,7 @@ function rand(F::iGMRF)
 
         Q = κ*W + e₁*e₁'
 
-    elseif F.rankDeficiency == 3
+    elseif F.rankDeficiency == 2
```

Comment:
Was this a bug?

Reply:
No not an error. A field or order 2 has a rank deficiency of 3.